org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/parser/Parser.java

```diff
@@ -4571,6 +4574,40 @@ protected void consumeInsideCastExpressionLL1WithBounds() {
 protected void consumeInsideCastExpressionWithQualifiedGenerics() {
 	// InsideCastExpressionWithQualifiedGenerics ::= $empty
 }
+protected void consumeTemplateExpressionWithPrimary() {
+	Expression expression = this.expressionStack[this.expressionPtr--];
+	StringTemplate template;
+	if (expression instanceof StringLiteral) {
+		template = new StringTemplate(new StringLiteral[] {(StringLiteral) expression},
+									new Expression[0], expression.sourceStart, expression.sourceEnd);
```

Comment:
Is there a test that verifies we do the right thing for a text block without any embedded expressions.